### ISSUE TITLE
Any documentation? How to integrate with current open source Apache Spark?

### ISSUE THREAD

**Issue by tfaruq** [SAP/HANAVora-Extensions#4]
Dear SAP,

Any documentation on this extension? How to integrate with current open source Apache Spark?

**Comment by burgerdev** [comment]
The HANAVora-Extensions are currently a standalone project that depends on spark. Running

```bash
mvn clean package
```

should prepare a package `dist/target/spark-sap-extensions-<ver>-dist.tar.gz`, which can be used at any site that has SPARK_HOME set. E.g., you can spawn a spark shell that includes the extensions by running

```
<package-dir>/bin/start-spark-shell.sh
```

Sorry to say that compiled package documentation is still on the ToDo list. If you are eager to dive into the codebase, you can perhaps try to follow the hint in `core/src/main/rootdoc.txt` and navigate the documentation starting from `SapSQLContext.scala`.

**Comment by ramnivas** [comment]
I wanted to try this route. However, `mvn clean package` is failing with unresolved dependencies (for what looks like sap extensions to Zeppelin):

```
$ mvn package
...
[ERROR] Failed to execute goal on project zeppelin: Could not resolve dependencies for project com.sap.spark:zeppelin:jar:1.3.69: The following artifacts could not be resolved: org.apache.zeppelin:zeppelin-interpreter:jar:0.5.6-incubating-sap-01, org.apache.zeppelin:zeppelin-spark:jar:0.5.6-incubating-sap-01: Failure to find org.apache.zeppelin:zeppelin-interpreter:jar:0.5.6-incubating-sap-01 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

How can I get the missing dependencies?

**Comment by ramnivas** [comment]
An update: I locally built Zeppelin 0.5.6 (without `-sap-01`, what I assume is a patch over the `v0.5.6` tag). After dropping `-sap-01` from Zeppelin dependencies in pom.xml, I could build the HANAVora-Extensions distribution (had to pass `-DskipTests`, since a few Selenium tests were failing). I am running into other issues, but at least I have a distribution.

It will be nice to have the Zeppelin 0.5.6-sap-01 version available somewhere, so that I can build without any hacks.

**Comment by stephankessler** [comment]
Hi,

unfortunately Zeppelin does not provide the zeppelin-interpreter jar on Maven central (http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.apache.zeppelin%22%20AND%20a%3A%22zeppelin-interpreter%22) thats why we built it internally on our own and put it into a local repo. Of course this is an issue - we are trying to resolve that internally.

The current workaround would be to put the zeppelin-interpreter in the local maven repo, and remove the sap-01 suffix.

Best,
Stephan

**Comment by ramnivas** [comment]
@stephankessler Thanks! I guess one thing that might help is to have the "sap-01" fork available on github. Then I would be able to `git clone` that and publish it to local maven. This way, I wouldn't need any change to the HAVAVORA-Extensions pom.xml.

**Comment by burgerdev** [comment]
The zeppelin issue was fixed in fab3d132, we currently use the public zeppelin version 0.6.1.

**Comment by burgerdev** [comment]
I added some documentation to start with, but the feature documentation is still work in progress.
